daml-assistant/daml-helper/src/DamlHelper.hs

```diff
@@ -346,45 +324,127 @@ runInit targetFolderM = do
         yamlConfig :: Y.Config
         yamlConfig = Y.setConfCompare fieldNameCompare Y.defConfig
 
-
-runNew :: FilePath -> String -> IO ()
-runNew targetFolder templateName = do
+-- | Create a DAML project in a new directory, based on a project template packaged
+-- with the SDK. Special care has been taken to avoid:
+--
+--   - project name / template name confusion: i.e. when a user passes a
+--   single argument, it should be the new project folder. But if the user
+--   passes an existing template name instead, we ask the user to be more
+--   explicit.
+--
+--   - creation of a project inside another project
+--   - creation of a project inside ano
+--
+runNew :: FilePath -> Maybe String -> IO ()
+runNew targetFolder templateNameM = do
     templatesFolder <- getTemplatesFolder
-    let templateFolder = templatesFolder </> templateName
+    let templateName = fromMaybe defaultProjectTemplate templateNameM
+        templateFolder = templatesFolder </> templateName
+        projectName = takeFileName (dropTrailingPathSeparator targetFolder)
+
+    -- Ensure template exists.
     unlessM (doesDirectoryExist templateFolder) $ do
         hPutStr stderr $ unlines
             [ "Template " <> show templateName <> " does not exist."
             , "Use `daml new --list` to see a list of available templates"
             ]
         exitFailure
+
+    -- Ensure project directory does not already exist.
     whenM (doesDirectoryExist targetFolder) $ do
         hPutStr stderr $ unlines
             [ "Directory " <> show targetFolder <> " already exists."
-            , "Please specify a new directory for creating a project."
+            , "Please specify a new directory, or use 'daml init' instead:"
+            , ""
+            , "    daml init " <> escapePath targetFolder
+            , ""
             ]
         exitFailure
+
+    -- Ensure user is not confusing template name with project name.
+    when (isNothing templateNameM && projectName == targetFolder) $ do
```

Comment:
Why do we need the `projectName == targetFolder` check here?

Reply:
My thinking is that if the user passes a targetFolder that isn't a straight up filename (i.e. contains path separators), then they probably know what they're doing.

Reply:
There's a world of difference between `daml new quickstart-java` and `daml new quickstart-java/` :-)

Reply:
Sounds very reasonable, would be nice to capture this thinking in a comment.

Reply:
Ok, i'll add it :)